transform-plugins/src/main/java/co/cask/hydrator/plugin/Normalize.java

```diff
@@ -0,0 +1,187 @@
+      String[] fieldNormalizingArray = config.fieldNormalizing.split(",");
+      for (String fieldNormalizing : fieldNormalizingArray) {
+        String[] fields = fieldNormalizing.split(":");
+        Preconditions.checkArgument(fields.length == 3, "Normalizing field '" + fields[0] + "' is invalid. " +
+          " Field Type and Field Value columns required.");
+        Preconditions.checkArgument(inputSchema.getField(fields[0]) != null, "Normalizing field '" + fields[0]
+          + "' not present in input schema.");
+      }
+    }
+    createOutputSchema();
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(outputSchema);
+  }
+
+  private void createOutputSchema() {
+    if (outputSchema != null) {
+      return;
+    }
+    //create output schema
+    List<Schema.Field> outputFields = Lists.newArrayList();
+    mappingFieldMap = new HashMap<String, String>();
+    String[] fieldMappingArray = config.fieldMapping.split(",");
+    for (String fieldMapping : fieldMappingArray) {
+      String[] mappings = fieldMapping.split(":");
+      Preconditions.checkArgument(mappings.length == 2, "Output schema field is missing for mapping field '" +
+        mappings[0] + "'.");
+      mappingFieldMap.put(mappings[0], mappings[1]);
+      outputFields.add(Schema.Field.of(mappings[1], Schema.of(Schema.Type.STRING)));
+    }
+
+    normalizeFieldMap = new HashMap<String, String>();
+    normalizeFieldList = Lists.newArrayList();
+    String[] fieldNormalizingArray = config.fieldNormalizing.split(",");
+
+    //Type and Value mapping for all normalize fields must be same, otherwise it is invalid.
+    //read type and value from first normalize fields which is used for validation.
+    String[] typeValueFields = fieldNormalizingArray[0].split(":");
+    String validTypeField = typeValueFields[1];
+    String validValueField = typeValueFields[2];
+
+    for (String fieldNormalizing : fieldNormalizingArray) {
+      String[] fields = fieldNormalizing.split(":");
+      Preconditions.checkArgument(mappingFieldMap.get(fields[0]) == null, "'" + fields[0] + "' cannot be use for " +
+        "both mapping as well as normalize fields.");
+      Preconditions.checkArgument(validTypeField.equals(fields[1]), "Type mapping is invalid for " +
+        "normalize field '" + fields[0] + "'. It must be same for all normalize fields.");
+      Preconditions.checkArgument(validValueField.equals(fields[2]), "Value mapping is invalid for " +
+        "normalize field '" + fields[0] + "'. It must be same for all normalize fields.");
+      normalizeFieldList.add(fields[0]);
+      normalizeFieldMap.put(fields[0] + NAME_KEY_SUFFIX, fields[1]);
+      normalizeFieldMap.put(fields[0] + VALUE_KEY_SUFFIX, fields[2]);
+    }
+    outputFields.add(Schema.Field.of(validTypeField, Schema.of(Schema.Type.STRING)));
+    outputFields.add(Schema.Field.of(validValueField, Schema.of(Schema.Type.STRING)));
+    outputSchema = Schema.recordOf("outputSchema", outputFields);
+  }
```

Comment:
@bhushankawadkar we should have a way to generate the schema from FE. Meaning use the service and integration with frontend with button to generate the schema or we should get user to enter output schema.